### ISSUE TITLE
Install ray on IBMZ i.e s390x

### ISSUE THREAD

**Issue by modassarrana89** [ray-project/ray#43353]
### Description

I don't see any s390x packages for ray.  

Still , i tried to install ray using pip command 
```
[root@m1305001 python]# pip install ray==2.4.0
ERROR: Could not find a version that satisfies the requirement ray==2.4.0 (from versions: none)
ERROR: No matching distribution found for ray==2.4.0
```

**Comment by modassarrana89** [comment]
> @modassarrana89 did you try latest ray?

Yes , I have . I am currently looking at the ray files to support s390x. It would be great if you can guide me what all file should i look apart from docker image 
```
ubuntu/focal
└── base-deps:cpu
    └── ray-deps:cpu
        └── ray:cpu
            └── ray-ml:cpu
```

**Comment by modassarrana89** [comment]
ray workspace is still pointing to s390x. 
```
[root@m1305001 ray]# git diff WORKSPACE 
diff --git a/WORKSPACE b/WORKSPACE
index 2283b4aaf8..1ee491fef5 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -32,9 +32,9 @@ hedron_compile_commands_setup()
 
 http_archive(
     name = "rules_python",
-    sha256 = "94750828b18044533e98a129003b6a68001204038dc4749f40b195b24c38f49f",
-    strip_prefix = "rules_python-0.21.0",
-    url = "https://github.com/bazelbuild/rules_python/releases/download/0.21.0/rules_python-0.21.0.tar.gz",
+    sha256 = "c68bdc4fbec25de5b5493b8819cfc877c4ea299c0dcb15c244c5a00208cde311",
+    strip_prefix = "rules_python-0.31.0",
+    url = "https://github.com/bazelbuild/rules_python/releases/download/0.31.0/rules_python-0.31.0.tar.gz",
 )
```
I have changed , but still it is taking older version. do i have to change somewhere else as well

**Comment by anyscalesam** [comment]
@aslonnie can you advise here?

**Comment by aslonnie** [comment]
upgrading the bazel rules would be pretty complicated given how ray uses bazel today, it will likely be intertwined with protobuf and grpc rules upgrade, and even if we managed to upgrade that, it will not be trivial to support a new architecture.

**Comment by anyscalesam** [comment]
@modassarrana89 could we connect over Slack more about your use case? Ask for invite here: https://docs.google.com/forms/d/e/1FAIpQLSfAcoiLCHOguOm8e7Jnn-JJdZaCxPGjgVCvFijHB5PLaQLeig/viewform

**Comment by modassarrana89** [comment]
@anyscalesam I am already on slack. Lets connect today

**Comment by modassarrana89** [comment]
> upgrading the bazel rules would be pretty complicated given how ray uses bazel today, it will likely be intertwined with protobuf and grpc rules upgrade, and even if we managed to upgrade that, it will not be trivial to support a new architecture.

I will be working on supporting ray on s390x architecture.

**Comment by modassarrana89** [comment]
@anyscalesam @aslonnie  Kserve is depenedent on Ray[serve] and we need to deploy kserve on s390x. Thats why we need s390x support on ray & its trivial for us . Please guide me regarding the changes which might impact protobuf & grpc.
